sw/device/silicon_creator/manuf/base/perso_tlv_data.c

```diff
@@ -85,52 +73,69 @@ rom_error_t perso_tlv_get_cert_obj(const uint8_t *buf, size_t ltv_buf_size,
   return kErrorOk;
 }
 
-status_t perso_tlv_prepare_cert_for_shipping(const char *name,
-                                             bool needs_endorsement,
-                                             const void *cert_body,
-                                             size_t cert_size,
-                                             perso_blob_t *pb) {
+rom_error_t perso_tlv_cert_obj_build(const char *name, bool needs_endorsement,
+                                     const uint8_t *cert, size_t cert_size,
+                                     uint8_t *buf, size_t *buf_size) {
   perso_tlv_object_header_t obj_header = 0;
   perso_tlv_cert_header_t cert_header = 0;
-  size_t name_len;
-  size_t obj_len;
-  size_t wrapped_len;
+  size_t obj_size;
+  size_t wrapped_cert_size;
 
   // Compute the name length (strlen() is not available).
-  name_len = 0;
+  size_t name_len = 0;
   while (name[name_len])
     name_len++;
   if (name_len > kCrthNameSizeFieldMask)
-    return OUT_OF_RANGE();
+    return kErrorPersoTlvCertNameTooLong;
 
   // Compute the wrapped certificate object (cert header + cert data) and perso
   // LTV object sizes.
-  wrapped_len = sizeof(perso_tlv_cert_header_t) + name_len + cert_size;
-  obj_len = wrapped_len + sizeof(perso_tlv_object_header_t);
+  wrapped_cert_size = sizeof(perso_tlv_cert_header_t) + name_len + cert_size;
+  obj_size = wrapped_cert_size + sizeof(perso_tlv_object_header_t);
 
   // Check there is enough room in the buffer to store the perso LTV object.
-  if (obj_len > (sizeof(pb->body) - pb->next_free))
-    return OUT_OF_RANGE();
+  if (obj_size > *buf_size)
+    return kErrorPersoTlvOutputBufTooSmall;
 
   // Setup the perso LTV object header.
   if (needs_endorsement) {
```

Comment:
Using this option to set object type is not practiable for CWT.
Maybe we'll need one more parameter?

Reply:
yeah lets add in follow up PR that adds code for CWT support